src/DataGrid.tsx

```diff
@@ -686,10 +686,16 @@ function DataGrid<R, SR, K extends Key>(
     if (enableEditor && isCellEditable(position)) {
       const row = rows[position.rowIdx] as R;
       setSelectedPosition({ ...position, mode: 'EDIT', key: null, row, originalRow: row });
-    } else {
+    } else if (
+      selectedPosition.mode !== 'SELECT' ||
```

Comment:
Can we add a test?

Reply:
How? 🤔

Reply:
If we do not call `onSelectedCellChange` then we can use it to test this change